app/src/lib/stores/app-store.ts

```diff
@@ -3347,44 +3348,89 @@ export class AppStore extends TypedBaseStore<IAppState> {
-      let branchToCheckout = branchesState.defaultBranch
-
-      // If the default branch is null, use the most recent branch excluding the branch
-      // the branch to delete as the branch to checkout.
-      if (branchToCheckout === null) {
-        let i = 0
-
-        while (i < branchesState.recentBranches.length) {
-          if (branchesState.recentBranches[i].name !== branch.name) {
-            branchToCheckout = branchesState.recentBranches[i]
-            break
-          }
-          i++
-        }
-      }
+      const branchToCheckout = this.getBranchToCheckoutAfterDelete(branch, r)
+      const gitStore = this.gitStoreCache.get(r)
 
-      if (branchToCheckout === null) {
-        throw new Error(
-          `It's not possible to delete the only existing branch in a repository.`
+      if (branchToCheckout != null) {
+        await gitStore.performFailableOperation(() =>
+          checkoutBranch(r, account, branchToCheckout)
         )
       }
 
-      const nonNullBranchToCheckout = branchToCheckout
-      const gitStore = this.gitStoreCache.get(r)
+      await gitStore.performFailableOperation(() => {
+        if (branch.type === BranchType.Remote) {
+          // Note: deleting a remote branch implementation not needed, yet.
+          return Promise.resolve(true)
+        }
+        return this.deleteLocalBranchAndUpstreamBranch(
+          repository,
+          branch,
+          account,
+          includeRemote
+        )
+      })
+
+      return this._refreshRepository(r)
+    })
+  }
+
+  /**
+   * Deletes the local branch. If the branch `includeRemote` is true, the
+   * upstream branch will be deleted also.
+   */
+  private async deleteLocalBranchAndUpstreamBranch(
+    repository: Repository,
+    branch: Branch,
+    account: IGitAccount | null,
+    includeRemote?: boolean
+  ): Promise<true> {
+    await deleteLocalBranch(repository, branch.name)
 
-      await gitStore.performFailableOperation(() =>
-        checkoutBranch(r, account, nonNullBranchToCheckout)
+    if (
+      includeRemote === true &&
+      branch.upstreamRemoteName !== null &&
+      branch.upstreamWithoutRemote !== null
+    ) {
+      await deleteRemoteBranch(
+        repository,
+        account,
+        branch.upstreamRemoteName,
+        branch.upstreamWithoutRemote,
+        branch.tip.sha
       )
-      await gitStore.performFailableOperation(() =>
-        deleteBranch(r, branch, account, includeRemote)
+    }
+    return true
+  }
+
+  private getBranchToCheckoutAfterDelete(
+    branchToDelete: Branch,
+    repository: Repository
+  ): Branch | null {
+    const { branchesState } = this.repositoryStateCache.get(repository)
+    const tip = branchesState.tip
+    const currentBranch = tip.kind === TipState.Valid ? tip.branch : null
+    // if current branch is not the branch being deleted, no need to switch
+    // branches
+    if (currentBranch != null && branchToDelete.name !== currentBranch.name) {
```

Comment:
Hmmm `!=` again… is that a mistake too or it's intended and I'm missing some Javscript/TypeScript magic here? 😳 
```suggestion
    if (currentBranch !== null && branchToDelete.name !== currentBranch.name) {
```

From what I see, `currentBranch` can only be either a `Branch` or `null`, so checking for `!=` shouldn't be necessary (it can never be `undefined`). Am I right? 🤔